platforms/core-configuration/model-core/src/main/java/org/gradle/internal/properties/bean/DefaultPropertyWalker.java

```diff
@@ -87,24 +90,52 @@ public void visitNestedUnpackingError(String qualifiedName, Exception e) {
 
             @Override
             public void visitLeaf(Object parent, String qualifiedName, PropertyMetadata propertyMetadata) {
-                PropertyValue cachedValue = new CachedPropertyValue(() -> propertyMetadata.getPropertyValue(parent), propertyMetadata.getDeclaredType().getRawType());
+                boolean isUpgradedProperty = propertyMetadata.isAnnotationPresent(ReplacesEagerProperty.class);
+                Class<?> propertyType = propertyMetadata.getDeclaredType().getRawType();
+                String propertyDisplayName = getPropertyDisplayName(parent, qualifiedName, propertyType, isUpgradedProperty);
+                PropertyValue cachedValue = new CachedPropertyValue(
+                    () -> propertyMetadata.getPropertyValue(parent),
+                    propertyMetadata.getDeclaredType().getRawType(),
+                    isUpgradedProperty,
+                    propertyDisplayName
+                );
                 PropertyAnnotationHandler handler = handlers.get(propertyMetadata.getPropertyType());
                 if (handler == null) {
                     throw new IllegalStateException("Property handler should not be null for: " + propertyMetadata.getPropertyType());
                 }
                 handler.visitPropertyValue(qualifiedName, cachedValue, propertyMetadata, visitor);
             }
+
+            @Nullable
+            private String getPropertyDisplayName(Object parent, String qualifiedName, Class<?> propertyType, boolean isUpgradedProperty) {
+                if (isUpgradedProperty && parent instanceof Task && FileCollection.class.isAssignableFrom(propertyType)) {
```

Comment:
💭 For FileCollection we build a better display name so we get nicer display name:
Old:
```
changing property value of this file collection at execution time is deprecated
```

New:
```
changing property value of task ':show' property 'prop' at execution time is deprecated
```

Alternatively we can drop this from this PR and properly wire it in general for generated classes.

Reply:
Actually found a better solution by implementing `OwnedAware` for `DefaultConfigurableFileCollection`